dts/arm/st/g0/stm32g0b0.dtsi

```diff
@@ -119,3 +121,21 @@
 		status = "disabled";
 	};
 };
+
+&itline24_mux {
+	status = "okay";
+};
+
+&i2c2 {
+	interrupt-parent = < &itline24_mux >;
+	interrupts = <0 0>;
+};
+
+&itline26_mux {
+	status = "okay";
```

Comment:
Ok fine
If not activating IT line in `.dtsi`, does it make sense to keep IT line being set as IT parent, like following one, in the `.dtsi` ?

```dts
&i2c2 {
	interrupt-parent = < &itline24_mux >;
	interrupts = <0 0>;
};
```

Asking because it wouldn't work without IT line being activated in the board `.dts`
Do you suggest to move all this IT related settings in the board `.dts` ?
Or is it better to make an exception, and keep them in the `.dtsi`, like in `dts/riscv/openisa/rv32m1_ri5cy.dtsi` ?

Reply:
It would, if child-nodes aren't featuring any `status = "disabled"`
Which results in having them enabled anyway, even if not used`
Thus, `_sw_isr_table` is populated with syscfg ISR for each and every IT line
Setting up a device IT without using syscfg itline results in a shared IT between syscfg and device driver ISRs
(Hope it's clear enough)
I don't think it's the expected behavior
Maybe better not to change any IT setting in `dtsi`, and let the user update as-needed in board `.dts` ?

> make the IT lines child-nodes

Implemented locally, syscfg driver updated accordingly, working as expected
Still need to address [your 3rd comment](https://github.com/zephyrproject-rtos/zephyr/pull/82776#discussion_r1879589884) before pushing

Reply:
Updated accordingly
No enabling of IT line mux nodes in `.dtsi`. Removed IT 2nd level related settings in configuration files, as not applicable anymore
Created a sample code to show how to setup board `.dts` and project configuration to use SYSCFG IT lines